pkg/prometheus/statefulset.go

```diff
@@ -752,9 +752,8 @@ func makeStatefulSetSpec(p monitoringv1.Prometheus, c *operator.Config, shard in
 				},
 			)
 
-			// NOTE(bwplotka): As described in https://thanos.io/components/sidecar.md/ we have to turn off compaction of Prometheus
```

Comment:
Should we add this warning to the documentation as well?

Reply:
Which documentation do you mean? The race condition happens rarely in some edge cases.

Reply:
I meant if we should move this comment to the documentation for the newly added field so that people are aware of the potential problems that can arise when the feature is enabled. Even if it could happen rarely in some cases, I think there's value in making that obvious for users.

Reply:
Yes, that makes sense. I will add the comment to the document and let users know. Users can take their own risks and make decision.